src/services/inventoryService.ts

```diff
@@ -427,6 +464,21 @@ export const addMiscItems = (inventory: IInventoryDatabaseDocument, itemsArray:
     });
 };
 
+export const addShipDecorations = (inventory: IInventoryDatabaseDocument, itemsArray: IConsumable[] | undefined) => {
+    const { ShipDecorations } = inventory;
+
+    itemsArray?.forEach(({ ItemCount, ItemType }) => {
+        const itemIndex = ShipDecorations.findIndex(miscItem => miscItem.ItemType === ItemType);
+
+        if (itemIndex !== -1) {
+            ShipDecorations[itemIndex].ItemCount += ItemCount;
+            inventory.markModified(`ShipDecorations.${itemIndex}.ItemCount`);
```

Comment:
markModified is only needed for Schema.Mixed. 
It should not be used. Mongoose tracks changes automatically, if there is a schema for the path.

Reply:
I  just copied the existing code for addMiscItems etc.